pkg/feature/manifest.go

```diff
@@ -83,6 +105,22 @@ func (t *templateManifest) Process(data any) ([]*unstructured.Unstructured, erro
 	return convertToUnstructuredSlice(resources)
 }
 
+func (t *templateManifest) SetManaged(objs []*unstructured.Unstructured) []*unstructured.Unstructured {
+	if t.patch {
+		return objs
+	}
+	for _, obj := range objs {
+		objAnnos := obj.GetAnnotations()
+		if objAnnos == nil {
+			objAnnos = make(map[string]string)
+		}
+
+		objAnnos[annotations.ManagedByODHOperator] = annotationTrue
+		obj.SetAnnotations(objAnnos)
+	}
+	return objs
+}
```

Comment:
```suggestion
func (t *templateManifest) SetManaged(objects []*unstructured.Unstructured) []*unstructured.Unstructured {
	if !t.patch {
		markAsManaged(objects)
	}

	return objects
}

func markAsManaged(objs []*unstructured.Unstructured) {
	for _, obj := range objs {
		objAnnotations := obj.GetAnnotations()
		if objAnnotations == nil {
			objAnnotations = make(map[string]string)
		}

		objAnnotations[annotations.ManagedByODHOperator] = "true"
		obj.SetAnnotations(objAnnotations)
	}
}
```

Reply:
```golang
objAnnotations[annotations.ManagedByODHOperator] = "true"
```

This hints that it might be better to flatten `pkg/metadata/labels` and `pkg/metadata/annotations` to one `pkg/metadata` and group `Labels` and `Annotations` in anonymous structs. Perhaps a follow-up PR at some point.

Then it would become:

```golang
annotations[metadata.Annotations.ManagedByODHOperator] = "true"
```